components/infra-proxy-service/migrations/pipeline/utility.go

```diff
@@ -738,3 +715,70 @@ func createLocalUser(ctx context.Context, localUserClient local_user.UsersMgmtSe
 	}
 	return nil
 }
+
+// StoreUsers reads the Result struct and populate the users table
+func StoreUsers(ctx context.Context, st storage.Storage, localUserClient local_user.UsersMgmtServiceClient, res pipeline.Result) (pipeline.Result, error) {
+	var err error
+	var totalSucceeded, totalSkipped, totalFailed int64
+
+	for _, user := range res.ParsedResult.Users {
+		_, err = StoreUser(ctx, st, user, res.Meta.ServerID, localUserClient)
+		if err != nil {
+			totalFailed++
+			continue
+		}
+		if user.ActionOps == pipeline.Skip {
+			totalSkipped++
+			continue
+		}
+		totalSucceeded++
+	}
+	if len(res.ParsedResult.Users) == int(totalFailed) {
+		log.Errorf("Failed to migrate user for migration id %s : %s", res.Meta.MigrationID, err.Error())
+		res.ParsedResult.UsersCount.Counts(int(totalSucceeded), int(totalFailed), int(totalSkipped))
+		return res, err
+	}
+
+	// Set the count for total Skipped, Inserted and Deleted Users
+	res.ParsedResult.UsersCount.Counts(int(totalSucceeded), int(totalFailed), int(totalSkipped))
+
+	log.Info("Successfully completed the user migration phase for migration id: ", res.Meta.MigrationID)
+	return res, err
+}
+
+// StoreUser stores a single User into DB
+func StoreUser(ctx context.Context, st storage.Storage, user pipeline.User, serverID string, localUserClient local_user.UsersMgmtServiceClient) (pipeline.ActionOps, error) {
+	var actionTaken pipeline.ActionOps
+	var err error
+
+	storageUser := storage.User{
+		ServerID:            serverID,
+		InfraServerUsername: user.Username,
+		Connector:           user.Connector,
+		Email:               user.Email,
+		DisplayName:         user.DisplayName,
+		FirstName:           user.FirstName,
+		LastName:            user.LastName,
+		MiddleName:          user.MiddleName,
+		AutomateUserID:      user.AutomateUsername,
+	}
+	switch user.ActionOps {
```

Comment:
```suggestion
	switch actionTaken = user.ActionOps; user.ActionOps {
```

This will remove unnecessary redefining actionTaken